### PR TITLE
Fix incorrect "not an app dir" warning

### PR DESCRIPTION
**Pull request by tuncer**

### CODE REVIEW

File: src/rebar_deps.erl [hunk 1]
```diff
@@ -457,8 +457,19 @@ is_app_available(Config, App, VsnRegex, Path, _IsRaw = false) ->
                               {AppFile, {expected, App}, {has, OtherApp}}}}}
             end;
         false ->
-            ?WARN("Expected ~s to be an app dir (containing ebin/*.app), "
-                  "but no .app found.\n", [Path]),
+            case filelib:is_dir(Path) of
+                true ->
+                    %% Path is a directory, but it's not an app dir.
+                    ?WARN("Directory expected to be an app dir, "
+                          "but no .app found:~n~s~n", [Path]);
```
Comment (ferd) [inline src/rebar_deps.erl]: This error message makes it sound like the `.app` file should be at the root of the app dir, which is false. Sounds like keeping a mention of `ebin/*.app` would be useful.

Comment (tuncer) [inline src/rebar_deps.erl]: I changed it because we also look for `src/*.app.src[.script]`. Any suggestions for an improved message?

Comment (ferd) [inline src/rebar_deps.erl]: Maybe?
```erlang
?WARN("Directory expected to be an app dir, "
      "but no app file found in ebin/ or src/:~n~s~n", [Path]);
```

Comment (tuncer) [inline src/rebar_deps.erl]: Thanks, updated.

**Pull request merged**